### DISCUSSION TITLE
lockFileMaintenance

### DISCUSSION THREAD

**Discussion by dgomesb** [renovatebot/renovate#31920] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

gitlab - ghcr.io/renovatebot/renovate:latest

### Please tell us more about your question or problem

The [doc](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance) says `By default, lockFileMaintenance is disabled`, however branches `renovate/*-lockfile` are being created. I tried to manually disable it within my `default.json`, `renovate-config.js`, and `.gitlab-ci.yml` but no luck.

It seems the default `schedule` and `groupName` also aren't working properly. When enabled I can't [group](https://docs.renovatebot.com/configuration-options/#groupname) multiple upgrades together. Renovate is creating branches and MRs at any time.

```
  lockFileMaintenance: {
    enabled: true,
    rebaseWhen: "behind-base-branch",
    branchTopic: "lock-file-maintenance",
    commitMessageAction: "Lock file maintenance",
    groupName: "lock-file-maintenance",
    schedule: ["before 4am on monday"],
    prBodyDefinitions: {"Change": "All locks refreshed"}
  },
```
- when I use `"recreateWhen": "always"` I get a warn message to change it to: `rebaseWhen: "behind-base-branch"`

Thank you in advance.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Those branches with a suffix of `-lockfile` are not lockFileMaintenance branches. They are targeted dependency updates which are updating the lock file only, and not the entire lock file being updated.

Please try reading https://docs.renovatebot.com/configuration-options/#lockfilemaintenance

**Comment by dgomesb** [comment]
The branches renovate creates are related to the same lock file `.terraform.lock.hcl`.
![image](https://github.com/user-attachments/assets/82dd415f-2eeb-4429-bc65-1a02f7f80c91)
Is there a way to combine them in one MR?

**Reply by rarkins** [reply]
Yes, using packageRules
